crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -0,0 +1,581 @@
+                .collect::<Result<Vec<_>, Error>>()?,
+        };
+        Ok(ManifestList { version, entries })
+    }
+
+    /// Get the entries in the manifest list.
+    pub fn entries(&self) -> &[ManifestListEntry] {
+        &self.entries
+    }
+
+    /// Get the version of the manifest list.
+    pub fn version(&self) -> &FormatVersion {
+        &self.version
+    }
+}
+
+/// Entry in a manifest list.
+#[derive(Debug, PartialEq, Clone)]
+pub struct ManifestListEntry {
+    /// field: 500
+    ///
+    /// Location of the manifest file
+    manifest_path: String,
+    /// field: 501
+    ///
+    /// Length of the manifest file in bytes
+    manifest_length: i64,
+    /// field: 502
+    ///
+    /// ID of a partition spec used to write the manifest; must be listed
+    /// in table metadata partition-specs
+    partition_spec_id: i32,
+    /// field: 517
+    ///
+    /// The type of files tracked by the manifest, either data or delete
+    /// files; 0 for all v1 manifests
+    content: ManifestContentType,
+    /// field: 515
+    ///
+    /// The sequence number when the manifest was added to the table; use 0
+    /// when reading v1 manifest lists
+    sequence_number: i64,
+    /// field: 516
+    ///
+    /// The minimum data sequence number of all live data or delete files in
+    /// the manifest; use 0 when reading v1 manifest lists
+    min_sequence_number: i64,
+    /// field: 503
+    ///
+    /// ID of the snapshot where the manifest file was added
+    added_snapshot_id: i64,
+    /// field: 504
+    ///
+    /// Number of entries in the manifest that have status ADDED, when null
+    /// this is assumed to be non-zero
+    added_data_files_count: Option<i32>,
+    /// field: 505
+    ///
+    /// Number of entries in the manifest that have status EXISTING (0),
+    /// when null this is assumed to be non-zero
+    existing_data_files_count: Option<i32>,
+    /// field: 506
+    ///
+    /// Number of entries in the manifest that have status DELETED (2),
+    /// when null this is assumed to be non-zero
+    deleted_data_files_count: Option<i32>,
+    /// field: 512
+    ///
+    /// Number of rows in all of files in the manifest that have status
+    /// ADDED, when null this is assumed to be non-zero
+    added_rows_count: Option<i64>,
+    /// field: 513
+    ///
+    /// Number of rows in all of files in the manifest that have status
+    /// EXISTING, when null this is assumed to be non-zero
+    existing_rows_count: Option<i64>,
+    /// field: 514
+    ///
+    /// Number of rows in all of files in the manifest that have status
+    /// DELETED, when null this is assumed to be non-zero
+    deleted_rows_count: Option<i64>,
+    /// field: 507
+    /// element_field: 508
+    ///
+    /// A list of field summaries for each partition field in the spec. Each
+    /// field in the list corresponds to a field in the manifest file’s
+    /// partition spec.
+    partitions: Option<Vec<FieldSummary>>,
```

Comment:
Remove the `Option`? I think empty vec is good enough for `None`?

Reply:
I think this will cause a case which will change manifest list implicitly
1. read partitions: Some(vec![]) 
2. write partitions back: None
Same things happend in key_metadata

Will it cause wierd for user?

Reply:
Good point. But since it's optional, we don't need to write `None` in serialized format? We just need to skip serialization.